### PR TITLE
refactor: Organise EC2 pattern for clarity

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

The [`ec2-app.ts`](https://github.com/guardian/cdk/blob/main/src/patterns/ec2-app.ts) file is pretty hard to reason about, partly due to its length (>500 lines).

In this change we:
  - Move the `GuPlayApp` and `GuNodeApp` classes into their own file
  - Remove the `GuApplicationPorts` enum as it doesn't provide much value
  - Update the integration test project to use a pattern, mainly to check the import path is still `import { GuPlayApp } from @guardian/cdk`. That is, this change is a no-op.

## How to test

CI should continue to pass.

## How can we measure success?

Navigating through the repo is easier.

## Have we considered potential risks?

n/a

## Checklist

- [ ] I have listed any breaking changes, along with a migration path [^1]
- [ ] I have updated the documentation as required for the described changes [^2]

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?

### CODE REVIEW

File: src/patterns/ec2-app/base.test.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe("the GuEC2App pattern", function () {
   it("should produce a functional EC2 app with minimal arguments", function () {
     const stack = simpleGuStackForTesting();
     new GuEc2App(stack, {
-      applicationPort: GuApplicationPorts.Node,
+      applicationPort: 3000,
```
Comment (akash1810) [inline src/patterns/ec2-app/base.test.ts:34]: I actually think this is more explicit than before!

File: src/patterns/ec2-app/base.test.ts [hunk 2]
```diff
@@ -817,48 +817,4 @@ describe("the GuEC2App pattern", function () {
       ],
     });
   });
-
-  describe("GuNodeApp", () => {
```
Comment (akash1810) [inline src/patterns/ec2-app/base.test.ts:821]: These get moved into their own dedicated test file.

**Review by jacobwinch — APPROVED** [review]
👍

File: src/patterns/ec2-app/base.test.ts (hunk 2 above)
Comment (jacobwinch) [reply src/patterns/ec2-app/base.test.ts:821]: 👍  - I like the `framework` naming too!

Comment (akash1810) [reply src/patterns/ec2-app/base.test.ts:821]: > 👍 - I like the `framework` naming too!

Phew! I spent more time worrying about the clarity of this than I should have! Sure `Play` is a framework, but `Node`? That's a runtime, no? Couldn't think of anything better, and we might actually remove the classes later as they don't do much anyway 😂 .

**Pull request merged**